src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -114,6 +115,43 @@ def try_path_munging(
 
 @region_silo_endpoint
 class ProjectStacktraceLinkEndpoint(ProjectEndpoint):  # type: ignore
+    """
+    Sorts the code mapping config list based on precedence.
+    User generated code mappings are evaluated before Sentry generated code mappings.
+    Code mappings with more defined stack trace roots are evaluated before less defined stack trace
+    roots.
+
+    `configs`: The list of code mapping configs
+    """
+
+    def sort_code_mapping_configs(
+        self,
+        configs: List[RepositoryProjectPathConfig],
+    ) -> List[RepositoryProjectPathConfig]:
+        sorted_configs = []
+        regex = r"\w+"
+        inserted = False
+        for config in configs:
+            for index, sorted_config in enumerate(sorted_configs):
+                if (
+                    sorted_config.automatically_generated and not config.automatically_generated
+                ) or (
+                    sorted_config.automatically_generated == config.automatically_generated
+                    and re.match(
+                        sorted_config.stack_root + regex,
+                        config.stack_root,
+                    )
+                ):
+                    sorted_configs.insert(index, config)
+                    inserted = True
+                    break
+            if not inserted:
+                if config.automatically_generated:
+                    sorted_configs.insert(len(sorted_configs), config)
+                else:
+                    sorted_configs.insert(0, config)
```

Comment:
If the config does not match the previous conditions, insert the config at the back if it's Sentry generated or the front if it is user generated